doc/ast-traversal.mld

```diff
@@ -0,0 +1,111 @@
+{%html: <div style="display: flex; justify-content:space-between"><div>%}{{!"matching-code"}< Destructing AST nodes}{%html: </div><div>%}{{!"good-practices"}Good practices >}{%html: </div></div>%}
+
+{0 AST Traversals}
+
+The {{!Ppxlib.Parsetree}[Parsetree]} is a very complex type. Other {!Ppxlib} modules such as
+{{!Ppxlib_metaquot}[Metaquot]}, {{!Ppxlib.Ast_builder}[Ast_builder]} and {{!Ppxlib.Ast_pattern}[Ast_pattern]} help in generating and matching values,
+but only when the overall structure of the code is known in advance.
+
+For other use cases, such as extracting all identifiers, checking that a
+property is verified, or replacing all integer constants by something else,
+those modules cannot really help. All these examples relate with another kind
+of {{!Ppxlib.Parsetree}[Parsetree]} manipulations known as traversals.
+
+A traversal is a recursive function that will be called on a value, and recursively on all
+of its subvalues, combining the result in a certain way. For instance, {{!Stdlib.List.map}[List.map]} is a traversal of the
+[list] type. In the case of a [list], a map is very simple to write, but in the
+case of the long {{!Ppxlib.Parsetree}[Parsetree]} type, it is a lot of boilerplate code! Fortunately,
+{{!Ppxlib}[ppxlib]} provides a way to ease this.
```

Comment:
I switched the section on "writing traverses" and the one on the different kind of traverses, to make the example appear early enough so that experienced reader can jump right to it.

Do you think it is sufficient?

Reply:
If you think it is, I'm pretty sure it is as well. We can always polish things later if we want to. Thanks for improving it!